### ISSUE TITLE
Mutations with fetchPolicy: 'no-cache' still trigger refetchQueries

### ISSUE THREAD

**Issue by bendenoz** [apollographql/apollo-client#10238]
**Intended outcome:**

I'm trying to execute mutations without triggering cache reconciliation because I have a lot of data in the cache, which can take up to >5secs.

**Actual outcome:**

even with fetchPolicy: 'no-cache', I see refetchQueries triggered.

**How to reproduce the issue:**

Load > 8000 objects in the cache in one big list query, trigger a mutation on unrelated objects with fetchPolicy: 'no-cache'

**Versions**

3.7.0

I believe the problem is in https://github.com/apollographql/apollo-client/blob/31fc8dfa22ca3074e95e521d28eba0544373ed8c/src/core/QueryManager.ts#L402
The test currently line 402 checks if refetchQueries exists, and since it appears to be an array it's always true, i guess it should be `mutation.refetchQueries.length > 0` 

My current workaround it to pass false `refetchQueries: false` to the mutate call but it fails the typescript constraint

**Comment by MacroMackie** [comment]
This has affected my team as well.

We have a note editor which periodically triggers api requests that look like this:
```
client.mutate({
  mutation: UpdateNoteMutation,
  fetchPolicy: "no-cache",
  variables: {
    input: {
      noteId,
    },
  },
});
```

The mutation triggers cache reconciliation, which can cause latency spikes when users are typing in our editor.

Our workaround has been to modify it at the "client defaultOptions" level, but definitely not ideal:
```
const newClient = new ApolloClient({
  ...
  defaultOptions: {
    mutate: {
      // ApolloClient's QueryManager updates the cache if this is undefined because it's replaced
      // with []. This will circumvent that issue.
      // See: https://github.com/apollographql/apollo-client/issues/10238
      //
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      refetchQueries: false as any,
    },
  },
});
```

I'm happy to share more info if that would be helpful!

**Comment by jessgoldq4** [comment]
@bendenoz is this still an issue in the latest version of `apollo-client`?

**Comment by druvisc** [comment]
You don't even need `refetchQueries` for automatic cache reconciliation to trigger (even on queries with `fetchPolicy: "network-only"`): https://github.com/apollographql/apollo-client/issues/11219#issuecomment-1765914610.

The only workaround is to set `nextFetchPolicy: 'standby'` on the queries so you can control when the data is re-fetched.

**Comment by vladar** [comment]
I've opened a PR for a quick fix: #11515. Curious if you see any pitfals with this simple fix?
